tests_func/load/mongo_load.go

```diff
@@ -0,0 +1,247 @@
+}
+
+func mongoRunCommandOp(client *mongo.Client, op *MongoRunCommandOp) {
+	db := client.Database(op.DbName)
+	var result bson.M
+	err := db.RunCommand(contextByName[op.Ctx], op.Doc).Decode(&result)
+	if op.MakeLogs {
+		if err != nil {
+			tracelog.InfoLogger.Printf("cannot execute runCommand: %+v", err)
+		} else {
+			tracelog.InfoLogger.Printf("Successful execution of runCommand with %+v argument", op.Doc)
+		}
+	}
+}
+
+func (mw *MongoWorker) mongoRunCommandOp(client *mongo.Client, op *MongoRunCommandOp) {
+	mw.fs = append(mw.fs, func() {
+		mongoRunCommandOp(client, op)
+		<-mw.ch
+		mw.wg.Done()
+	})
+}
+
+type MongoRunTransactionOp struct {
+	Ctx      string                       `json:"ctx"`
+	MakeLogs bool                         `json:"makelogs"`
+	Cmds     []map[string]json.RawMessage `json:"cmds"`
+	callback func(client *mongo.Client) (interface{}, error)
+}
+
+func parseTransactionOp(cd map[string]json.RawMessage) (*MongoRunTransactionOp, error) {
+	var res MongoRunTransactionOp
+	bmrco, err := json.Marshal(cd)
+	if err != nil {
+		return nil, fmt.Errorf("cannot parse opts for commad: %+v", err)
+	}
+	err = json.Unmarshal(bmrco, &res)
+	if err != nil {
+		return nil, fmt.Errorf("cannot parse opts for commad: %+v", err)
+	}
+	var fns []func(client *mongo.Client)
+	for _, cmd := range res.Cmds {
+		commandDoc, err := parseCommandOp(cmd)
+		if err != nil {
+			return nil, err
+		}
+		fns = append(fns, func(client *mongo.Client) {
+			mongoRunCommandOp(client, commandDoc)
+		})
+	}
+	res.callback = func(client *mongo.Client) (interface{}, error) {
+		for _, f := range fns {
+			f(client)
+		}
+		return nil, nil
+	}
+	return &res, nil
+}
+
+func (mw *MongoWorker) mongoRunTransaction(client *mongo.Client, op *MongoRunTransactionOp) {
+	mw.fs = append(mw.fs, func() {
+		session, err := client.StartSession()
+		if op.MakeLogs {
+			if err != nil {
+				tracelog.InfoLogger.Printf("Cannot start session: %+v", err)
+			} else {
+				tracelog.InfoLogger.Printf("Successful execution of transaction with %+v argument", op.Cmds)
+			}
+		}
+		if err != nil {
+			<-mw.ch
+			mw.wg.Done()
+			return
+		}
+		defer session.EndSession(context.Background())
+		_, err = session.WithTransaction(contextByName[op.Ctx], func(sessCtx mongo.SessionContext) (interface{}, error) {
+			return op.callback(client)
+		})
+		if op.MakeLogs {
+			if err != nil {
+				tracelog.InfoLogger.Printf("Failed execution of transaction: %+v", err)
+			} else {
+				tracelog.InfoLogger.Printf("Successful execution of transaction with %+v argument", op.Cmds)
+			}
+		}
+		<-mw.ch
+		mw.wg.Done()
+	})
+}
+
+func (mw *MongoWorker) Run() {
+	for _, f := range mw.fs {
+		mw.ch <- struct{}{}
+		mw.wg.Add(1)
+		go f()
+	}
+	mw.wg.Wait()
+}
+
+func multiplyInArray(value string, cnt int) []string {
+	var res []string
+	for cnt > 0 {
+		res = append(res, value)
+		cnt--
+	}
+	return res
+}
+
+func main() {
+
+	err := generatePatronsFromFile("./tests_func/load/mongo_load_config.json")
```

Comment:
so we're developing 2 different tools here: "bullet generator" and "load generator"
they should be separated (different files and binaries), let's move it to different files
after that we need to make convenient interface between them:
-   "bullet generator" writes bullets to stdout or to file
-   "load generator" reads bullets from stdin or from file

Reply:
This is an example of usage now, I think I'll remove it when it is ready to merge with master, because there will be usage of this somewhere else.